src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2031,7 +2031,21 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 	workerExtendedOpNode->hasWindowFuncs = originalOpNode->hasWindowFuncs;
 	workerExtendedOpNode->windowClause = originalOpNode->windowClause;
 
-	if (!queryHasAggregates)
+	/*
+	 * we are only pushing down distinct on, and limit (along with order by)
+	 * if it distinct on clause is on bare column. If it has expression, aggregates
+	 * or function calls, they are evaluated at worker query and returned to master
+	 * to perform necessary distinct on/order by/limit operations.
+	 */
+	if (enableLimitPushdown && originalOpNode->distinctClause && groupClauseList != NIL)
```

Comment:
`groupClauseList != NIL` could we remove that since it is already checked in `IsGroupBySubsetOfDistinct` ?

Reply:
```IsGroupBySubsetOfDistinct()``` returns false if group by clause is empty which causes setting orderby/limit pushdown flags to false. We do not want that.